internal/pkg/testutil/kube.go

```diff
@@ -0,0 +1,447 @@
+									Value: "test",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// GetStatefulSet provides statefulset for testing
+func GetStatefulSet(namespace string, statefulsetName string) *v1_beta1.StatefulSet {
+	return &v1_beta1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      statefulsetName,
+			Namespace: namespace,
+			Labels:    map[string]string{"firstLabel": "temp"},
+			Annotations: map[string]string{
+				common.ConfigmapUpdateOnChangeAnnotation: statefulsetName,
+				common.SecretUpdateOnChangeAnnotation:    statefulsetName},
+		},
+		Spec: v1_beta1.StatefulSetSpec{
+			UpdateStrategy: v1_beta1.StatefulSetUpdateStrategy{
+				Type: v1_beta1.RollingUpdateStatefulSetStrategyType,
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"secondLabel": "temp"},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Image: "tutum/hello-world",
+							Name:  statefulsetName,
+							Env: []v1.EnvVar{
+								{
+									Name:  "BUCKET_NAME",
+									Value: "test",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// GetConfigmap provides configmap for testing
+func GetConfigmap(namespace string, configmapName string, testData string) *v1.ConfigMap {
+	return &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      configmapName,
+			Namespace: namespace,
+			Labels:    map[string]string{"firstLabel": "temp"},
+		},
+		Data: map[string]string{"test.url": testData},
+	}
+}
+
+// GetConfigmapWithUpdatedLabel provides configmap for testing
+func GetConfigmapWithUpdatedLabel(namespace string, configmapName string, testLabel string, testData string) *v1.ConfigMap {
+	return &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      configmapName,
+			Namespace: namespace,
+			Labels:    map[string]string{"firstLabel": testLabel},
+		},
+		Data: map[string]string{"test.url": testData},
+	}
+}
+
+// GetSecret provides secret for testing
+func GetSecret(namespace string, secretName string, data string) *v1.Secret {
+	return &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: namespace,
+			Labels:    map[string]string{"firstLabel": "temp"},
+		},
+		Data: map[string][]byte{"test.url": []byte(data)},
+	}
+}
+
+// GetSecretWithUpdatedLabel provides secret for testing
+func GetSecretWithUpdatedLabel(namespace string, secretName string, label string, data string) *v1.Secret {
+	return &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: namespace,
+			Labels:    map[string]string{"firstLabel": label},
+		},
+		Data: map[string][]byte{"test.url": []byte(data)},
+	}
+}
+
+// VerifyDeploymentUpdate verifies whether deployment has been updated with environment variable or not
+func VerifyDeploymentUpdate(client kubernetes.Interface, namespace string, name string, envarPostfix string, shaData string, annotation string) bool {
```

Comment:
All these verification methods are identical. Use a func for fetching containers for each please and use 1 method